tests/test_liver.py

```diff
@@ -0,0 +1,370 @@
+    assert isinstance(liver_prediction, torch.Tensor)
+    assert liver_prediction.shape[0] == 1
+    assert liver_prediction.shape[1] == 1  # number of channels
+    assert (
+        liver_prediction.shape[2:]
+        == nib.load(config.test["test_volume"]).get_fdata().shape
+    )  # original volume dimension
+
+    prediction = liver_prediction.cpu()
+    prediction = prediction.numpy()
+    final_path = path
+    true = np.load(final_path)
+    assert np.allclose(prediction, true, 0.01)
+
+
+@pytest.fixture
+def liver_obj():
+    config.transforms["transformation_size"] = [64, 64]
+    liver_obj = LiverSegmentation(modality="CT", inference="3D")
+    return liver_obj
+
+
+def test_predict_2dto3d(liver_obj):
+    """
+    Tests predict_2dto3d function.
+    verifies the functionality by performing predictions on a 3D volume (with size 64,64,...)
+    """
+
+    prediction = liver_obj.predict_2dto3d(config.test["test_volume"])
+    assert isinstance(prediction, torch.Tensor)
+    assert prediction.shape[0] == 1
+    assert prediction.shape[1] == 1  # number of channels
+
+    assert (
+        prediction.shape[2:] == nib.load(config.test["test_volume"]).get_fdata().shape
+    )  # original volume dimension
+
+    assert torch.min(prediction) >= 0
+    assert torch.max(prediction) <= 1
+
+
+@pytest.fixture
+def liver_object_sw():
+    liver_object_sw = LiverSegmentation(modality="CT", inference="sliding_window")
+    return liver_object_sw
+
+
+def test_predict_sliding_window(liver_object_sw):
+    """
+    Tests predict_sliding_window function.
+    verifies the functionality by performing predictions on a 3D volume (with size 64,64,...)
+    """
+    prediction = liver_object_sw.predict_sliding_window(
+        volume_path=config.test["test_volume"]
+    )
+    assert isinstance(prediction, torch.Tensor)
+    assert prediction.shape[0] == 1
+    assert prediction.shape[1] == 1  # number of channels
+
+    assert (
+        prediction.shape[2:] == nib.load(config.test["test_volume"]).get_fdata().shape
+    )  # original volume dimension
+
+    assert torch.min(prediction) >= 0
+    assert torch.max(prediction) <= 1
+
+
+def test_test_sliding_window(liver_object_sw):
+    """
+    Tests test_sliding_window function.
+    verifies the functionality by comparing loss with true refrence loss on the same volume
+    """
+    test_dataloader = DataLoader(
+        dataset_path=config.test["test_folder"],
+        batch_size=config.training["batch_size"],
+        train_transforms=liver_object_sw.train_transform,
+        test_transforms=liver_object_sw.test_transform,
+        num_workers=0,
+        pin_memory=False,
+        test_size=1,
+        mode= "2D",
+        shuffle=config.training["shuffle"],
+    )
+    test_dataload = test_dataloader.get_testing_data()
+    test_loss, test_metric = liver_object_sw.test_sliding_window(
+        dataloader=test_dataload
+    )
+
+    assert round(test_loss,1) == 0.9
```

Comment:
I think the specified number of decimals to round to is very small, increase it if possible to increase test accuracy.